src/main/java/com/javatemplate/domain/user/UserService.java

```diff
@@ -37,6 +41,12 @@ public User create(final User user) {
         return userStore.create(user);
     }
 
+    public UserDetails loginWithFacebook(final org.springframework.social.facebook.api.User userLogin) {
+        final Optional<User> userFoundOptional = userStore.findByEmail(userLogin.getEmail());
+
+        return jwtUserDetailsService.loadUserByUsername(userFoundOptional.get().getUsername());
+    }
+
```

Comment:
Logic chỗ này hoàn toàn không liên quan gì đến facebook

```suggestion
    public UserDetails loginWithFacebook(final String facebookToken) {
       // Logic copy bên controller qua đây
        return loginByUsername(email);
    }


   public UserDetails loginByUsername(final String username) {
        return userStore.findByEmail(username)
                         .map(UserDetails::new)
                         .orElseThrow(...);
    }
```

Reply:
em đổi sang như thế này được không anh? Em làm theo cách anh nhưng mà có lỗi hoặc là em tạo thêm 1 method loadUserByEmail nữa anh.
![image](https://user-images.githubusercontent.com/87653000/228446627-de67f595-035c-4c94-aca4-df40e1ac1931.png)

Reply:
![image](https://user-images.githubusercontent.com/87653000/228446800-437ed9f4-f2a6-474c-a42f-29a8a17789ef.png)